pkg/provider/provider.go

```diff
@@ -586,25 +571,28 @@ func (k *kubeProvider) Diff(
 	}
 
 	var patch []byte
+	var isInputPatch bool
+	var patchBase *unstructured.Unstructured
+
 	tryDryRun := supportsDryRun && oldInputs.GroupVersionKind().String() == gvk.String()
 	if tryDryRun {
-		var live *unstructured.Unstructured
-		patch, live, err = k.dryRunPatch(oldInputs, newInputs)
+		patch, patchBase, err = k.dryRunPatch(oldInputs, newInputs)
 
-		// Fall back to local patch.
+		// Fall back to input patch.
 		se, isStatusError := err.(*errors.StatusError)
 		if isStatusError && se.Status().Code == 400 &&
 			(se.Status().Message == "the dryRun alpha feature is disabled" ||
 				se.Status().Message == "the dryRun beta feature is disabled") {
 
-			patch, err = k.localPatch(oldInputs, newInputs, oldLiveState)
-		} else {
-			oldLiveState = live
+			isInputPatch = true
+			patch, err = k.inputPatch(oldInputs, newInputs)
+			patchBase = oldInputs
 		}
 	} else {
-		patch, err = k.localPatch(oldInputs, newInputs, oldLiveState)
+		isInputPatch = true
+		patch, err = k.inputPatch(oldInputs, newInputs)
+		patchBase = oldInputs
```

Comment:
Just to protect future, tiny-brained Alex, can we please add a `.V(1)` log in the appropriate spot after this block to indicating which patch type was used to calculate the diff? I'd rather have that information than not when users file issues.

Reply:
Yes.

Reply:
Done.